src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -277,19 +332,22 @@ private string CleanWhitespace(string rawText)
             return string.Concat(rawText.Where(c => !char.IsWhiteSpace(c)));
         }
 
-        private string ConstructUserSettingsJson()
+        private string ConstructSettingsJson(List<string> permittedFileNames = null)
```

Comment:
nit: params string[] may be more convenient

Reply:
The `params` keyword is still missing; if you add that, it will alleviate the need to check for null on this parameter and the need to manually create a new array at each callsite

Reply:
Oh, my mistake - I wasn't aware of the `params` keyword and just read it as Wiktor suggesting I use string[] as my parameter type. I'll fix that.